snapcraft/tests/fake_servers/snapd.py

```diff
@@ -51,6 +57,14 @@ def _handle_snaps(self, parsed_url):
         elif parsed_url.path.endswith('/fake-snap-edge'):
             status_code = 200
             params = {'channel': 'edge'}
+        elif (parsed_url.path in self.snaps_not_found and
+              parsed_url.path.split('/')[-1] in self.installed_snaps):
+            # XXX when the snaps end point fails, the snap is installed and
+            # it needs a revision the next time the same endpoint is called.
+            status_code = 200
+            params = {'channel': 'dummy', 'revision': 'dummy'}
+        else:
+            self.snaps_not_found.append(parsed_url.path)
```

Comment:
I'm finding it difficult to follow how `snaps_not_found` works...

Reply:
I don't have a lot of options here, because I need the first reply to be different to the second. But I'm experimenting with a few things that I will propose in a follow up branch.